tests/naive.py

```diff
@@ -1716,3 +1716,74 @@ def _total_diagonal_ndists(tile_lower_diag, tile_upper_diag, tile_height, tile_w
         )
 
     return total_ndists
+
+
+def stump_topk(T_A, m, T_B=None, exclusion_zone=None, k=1):
+    """
+    Traverse distance matrix along the diagonals and update the top-k
+    nearest neigbors matrix profile and matrix profile indices
+    """
+    if T_B is None:  # self-join:
+        ignore_trivial = True
+        distance_matrix = np.array(
+            [distance_profile(Q, T_A, m) for Q in core.rolling_window(T_A, m)]
+        )
+        T_B = T_A.copy()
+    else:
+        ignore_trivial = False
+        distance_matrix = np.array(
+            [distance_profile(Q, T_B, m) for Q in core.rolling_window(T_A, m)]
+        )
+
+    distance_matrix[np.isnan(distance_matrix)] = np.inf
```

Comment:
After we've computed the full distance matrix, I think we would "naively":

1. Apply an exclusion zone to the distance matrix if it is a self-join
2. Perform a row-wise argsort and take the top-k columns 

Return the corresponding matrix profile data structure. Our naive approach does NOT need to follow STUMPY's diagonal traversal. Now, there is one complication and that is what happens when there is a tie in distances? Which index is recorded by `stumpy.stump`?  We'll need to deal with that.

Reply:
> Now, there is one complication and that is what happens when there is a tie in distances? Which index is recorded by `stumpy.stump`? We'll need to deal with that.

In the current design, we do NOT update matrix profile index when there is a tie. Therefore, since we traverse diagonally,  `stump` returns the lower index when there is a tie. I think `np.argsort` is the same and so there should be no problem (I hope I got your point correctly) I will investigate this to make sure of it.

Reply:
Yes, you got my point precisely. Just something to keep in mind if we see that the matrix profile distances are the same but the indices are different.